src/index.ts

```diff
@@ -19,3 +25,15 @@ export const recursiveReaddir = (dir: string): asyncIt.AsyncIterableEx<string> =
         }
     )
 
+export const recursiveRmdir = async (dir: string): Promise<void> => {
```

Comment:
ther eis already a de facto standard library for handlings FS stuff, I would not reinvent the wheel and use whats proven in the field https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md

Reply:
TypeScript definitions are two versions behind 
https://www.npmjs.com/package/@types/fs-extra 7.0
https://www.npmjs.com/package/fs-extra 5.0

Reply:
And they don't have recursive readdir with async iterator. BTW, it has the same problem with type script definitions that to write correct TypeScript definition is really hard and require much more work and testing than to write a new library for such small functions.